### DISCUSSION TITLE
Replace lodash with native JS functions

### DISCUSSION THREAD

**Discussion by adamberecz** [vueform/vueform#45] [Ideas]
Replace lodash with native JS functions

**Comment by Aghaei-Dev** [comment]
witch one of their functions u want?

**Reply by adamberecz** [reply]
All of them eventually

**Reply by Aghaei-Dev** [reply]
:/

**Comment by rogerhendriks** [comment]
Great idea as lodash has quite some security issues

**Comment by jadrake75** [comment]
So there is a nifty eslint plugin I use in my project called [eslint-plugin-you-dont-need-lodash-underscore](https://www.npmjs.com/package/eslint-plugin-you-dont-need-lodash-underscore) which will spit out warnings and suggestions on replacements.  I LOVE lodash and have used it for well over 10 years but alas, training myself on how to work with arrays and objects using the latest and greatest techniques is the way to go

deepClone( ) --> structuredClone( ) seems to work, but if you have a ref( ) value you may need to call structuredClone(toRaw(object.value)) to actually get it to work in certain situations

**Reply by adamberecz** [reply]
Thanks for the recommendation. I'll keep it in mind when I get to this point.

**Comment by tomasvn** [comment]
[https://es-toolkit](https://es-toolkit.slash.page/)

If you even want to preserve lodash-like you could check this package or even check out this page [You don’t need Lodash](https://you-dont-need.github.io/You-Dont-Need-Lodash-Underscore/#/)
